test.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash --
+set -euo pipefail
+tmpdir=$(mktemp -d)
+
+case $0 in (/*) cd "${0%/*}/";; (*/*) cd "./${0%/*}";; (*) :;; esac
+(
+set -x
+truncate -s 0 dummy.img
+truncate -s 20GiB dummy.img
+sfdisk --force dummy.img < layout
+sfdisk --dump dummy.img > layout2
+loopdev=$(sudo losetup --nooverlap --find --sector-size 4096 --show -- dummy.img)
+trap 'sudo losetup -D -- "$loopdev"' EXIT
```

Comment:
-D is detach-all, you want -d with specific device

Reply:
Fixed.